kong/plugins/moesif/log.lua

```diff
@@ -136,55 +148,62 @@ function get_config_internal(conf)
+        end
 
-      -- Get governance rules
-      if (governance_rules_etags[hash_key] == nil or (conf["rulesETag"] ~= governance_rules_etags[hash_key])) then
-        gr_helpers.get_governance_rules(hash_key, conf)
-      end
+        -- Check if the governance rule is updated
+        local response_rules_etag = string.match(config_response, "Tag%s*:%s*(.-)\n")
+          if response_rules_etag ~= nil then
+          conf["rulesETag"] = response_rules_etag
+        end
 
-      if (response_body["user_rules"] ~= nil) then
-        entity_rules[hash_key]["user_rules"] = response_body["user_rules"]
-      end
-        
-      if (response_body["company_rules"] ~= nil) then
-          entity_rules[hash_key]["company_rules"] = response_body["company_rules"]
-      end
+        -- Hash key of the config application Id
+        local hash_key = string.sub(conf.application_id, -10)
 
-      if (conf["sample_rate"] ~= nil) and (response_body ~= nil) then
-        if (response_body["user_sample_rate"] ~= nil) then
-          conf["user_sample_rate"] = response_body["user_sample_rate"]
+        -- Create empty table for user/company rules
+        if entity_rules[hash_key] == nil then
+          entity_rules[hash_key] = {}
         end
 
-        if (response_body["company_sample_rate"] ~= nil) then
-          conf["company_sample_rate"] = response_body["company_sample_rate"]
+        -- Get governance rules
+        if (governance_rules_etags[hash_key] == nil or (conf["rulesETag"] ~= governance_rules_etags[hash_key])) then
+          gr_helpers.get_governance_rules(hash_key, conf)
         end
 
-        if (response_body["regex_config"] ~= nil) then
-          conf["regex_config"] = response_body["regex_config"]
+        if (response_body["user_rules"] ~= nil) then
+          entity_rules[hash_key]["user_rules"] = response_body["user_rules"]
+        end
+          
+        if (response_body["company_rules"] ~= nil) then
+            entity_rules[hash_key]["company_rules"] = response_body["company_rules"]
         end
 
-        if (response_body["sample_rate"] ~= nil) then 
-          conf["sample_rate"] = response_body["sample_rate"]
+        if (conf["sample_rate"] ~= nil) and (response_body ~= nil) then
+          if (response_body["user_sample_rate"] ~= nil) then
+            conf["user_sample_rate"] = response_body["user_sample_rate"]
+          end
+
+          if (response_body["company_sample_rate"] ~= nil) then
+            conf["company_sample_rate"] = response_body["company_sample_rate"]
+          end
+
+          if (response_body["regex_config"] ~= nil) then
+            conf["regex_config"] = response_body["regex_config"]
+          end
+
+          if (response_body["sample_rate"] ~= nil) then 
+            conf["sample_rate"] = response_body["sample_rate"]
+          end
+        end
+      else
+        if conf.debug then
+          ngx_log(ngx.DEBUG, "[moesif] raw config response is nil so could not decode it ")
```

Comment:
here is ok ..print config_response also

Reply:
okay, have added config_response to the log